### PR TITLE
Fix `argmin`/`argmax` `dtype` argument

### LINKED ISSUE #2595: Signatures and behaviors of `argmax` and `argmin` are incompatible with NumPy
First, for the signature: In CuPy the signature for `argmax()` and `argmin()` is 
```python
cupy.argm*(a, axis=None, dtype=None, out=None, keepdims=False)
```
But in NumPy it's 
```python
numpy.argm*(a, axis=None, out=None)
```
That is, `dtype` and `keepdims` should be removed.

Second, for the behavior: In CuPy the `axis` could be a tuple, but in Numpy it can only be an integer:
```python
>>> import cupy as cp
>>> a = cp.arange(60).reshape(3,4,5)
>>> a.argmax(axis=(0,1))
array([11, 11, 11, 11, 11], dtype=int64)
>>> 
>>> import numpy as np
>>> a = np.arange(60).reshape(3,4,5)
>>> a.argmax(axis=(0,1))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'tuple' object cannot be interpreted as an integer
```

I think both are easy to fix. Just add a few guards prior to calling the actual workhorses.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #2595
Requires #2870

### CODE REVIEW

File: cupy/core/_routines_statistics.pyx [hunk 1]
```diff
@@ -206,27 +206,27 @@ nanmax = create_reduction_func(
...
 cdef _argmin = create_reduction_func(
     'cupy_argmin',
-    ('?->q', 'B->q', 'h->q', 'H->q', 'i->q', 'I->q', 'l->q', 'L->q',
-     'q->q', 'Q->q',
-     ('e->q', (None, 'my_argmin_float(a, b)', None, None)),
-     ('f->q', (None, 'my_argmin_float(a, b)', None, None)),
-     ('d->q', (None, 'my_argmin_float(a, b)', None, None)),
-     ('F->q', (None, 'my_argmin_float(a, b)', None, None)),
-     ('D->q', (None, 'my_argmin_float(a, b)', None, None))),
+    tuple(['{}->{}'.format(d, r) for r in 'bhilq' for d in '?BhHiIlLqQ'])
```
Comment (leofang) [inline cupy/core/_routines_statistics.pyx]: I am a bit concerned with the order `'bhilq'`. Which return type is chosen when `argmin(..., dtype=None, ...)` is called? I thought we'd need to put `'q'` as the leading character to match NumPy default, but I could be wrong. (I don't have access now to test this PR.)

**Comment by leofang** [comment]
To close #2595, it'd be nice to document the API deviation (which is a good thing I think).

File: cupy/core/_routines_statistics.pyx (hunk 1 above)
Comment (niboshi) [reply cupy/core/_routines_statistics.pyx]: Thank you for pointing out! It seems you are correct. Some of the tests were failing.

**Comment by emcastillo** [comment]
Waiting till #2870 is merged

**Comment by niboshi** [comment]
#2870 merged.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**